subsys/net/ip/net_if.c

```diff
@@ -2864,6 +2864,16 @@ struct net_if_ipv6_prefix *net_if_ipv6_prefix_get(struct net_if *iface,
 out:
 	net_if_unlock(iface);
 
+	if (prefix != NULL) {
```

Comment:
You may want to do this before `net_if_unlock` to make sure the prefix is not changed in the middle of printing it.